### PR TITLE
Secret Management

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**

### CODE REVIEW

File: docs/docs/0.4.0/documentation/secret-management.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+```
+
+#### Access to the Secret API
+
+This API is exposed on Admin Server which means that by default it is only available on the same machine as the Control Plane.
+Consult [Accessing Admin Server from a different machine](security/#accessing-admin-server-from-a-different-machine) how to configure remote access.
+
+### Kubernetes
+
+On Kubernetes, Kuma leverages native [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/) mechanism.
```
Comment (subnetmarco) [inline docs/docs/0.4.0/documentation/secret-management.md:42]: "the native [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/) resource"

File: docs/docs/0.4.0/documentation/secret-management.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+data: dGVzdAo=
+
+$ kumactl apply -f secret.yaml
+
+$ kumactl get secrets
+MESH      NAME
+default   sample-secret 
+```
+
+#### Access to the Secret API
```
Comment (subnetmarco) [inline docs/docs/0.4.0/documentation/secret-management.md:35]: Where is the API? Is it an HTTP API?

**Comment by subnetmarco** [comment]
@jakubdyszkiewicz When applying a Secret like:

```yaml
type: Mesh
name: default
mtls:
  backends:
  - name: ca-1
    type: provided
    config:
      cert:
        secret: my-cert # name of the Kuma Secret
      key:
        secret: my-key # name of the Kuma Secret
```

are we checking that the secret does indeed exist? 

The same check should be made for Kubernetes.

File: docs/docs/0.4.0/documentation/secret-management.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+Use `kumactl` to manage secret like any other Kuma resource on Universal.
+
+```sh
+$ cat secret.yaml
+type: Secret
+mesh: default
+name: sample-secret
+data: dGVzdAo=
+
+$ kumactl apply -f secret.yaml
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply docs/docs/0.4.0/documentation/secret-management.md]: Then you end up in sensitive information in your bash history.

**Comment by jakubdyszkiewicz** [comment]
Yes, we validate if secrets exists. On both envs.

File: docs/docs/0.4.0/documentation/secret-management.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+### Universal
+
+The following snippet presents the format of Secret
+
+```yaml
+type: Secret
+name: sample-secret
+mesh: default
+data: dGVzdAo= # bytes encoded in Base64
```
[the remark this answers is not in the document]
Comment (jakubdyszkiewicz) [reply docs/docs/0.4.0/documentation/secret-management.md]: This is a little problematic on K8S https://kubernetes.io/docs/concepts/configuration/secret/
You can use `stringData` to apply secret in plaintext, but it is converted into base64 data anyways. See section:
"The stringData field is a write-only convenience field. It is never output when retrieving Secrets. For example, if you run the following command:"

**Pull request merged**